src/fitsviewer.py

```diff
@@ -504,9 +510,33 @@ def __init__(self, parent=None):
         self.active_plot_window = None
 
     def on_indices_toggled(self, state):
-
         self.ui.arrayIndices.setEnabled(state)
 
+    def create_analysis_template(self):
+        index = self.ui.sections.currentIndex()
+        #data_str=str(self.ui.sections.model().index(index.row(),index.column()).data())
+        template='''import numpy\nfrom matplotlib import pyplot as plt\nfrom astropy.io import fits\nhdul = fits.open("{fname}")\nprint(hdul.info())\n'''
+        template+='print(hdul[{index}].data.names)\n'
+        template+='fig=plt.figure()\nplt.plot(hdul[{index}].data)\nplt.show()'
+        fname=self.current_file if self.current_file else ''
+        row=index.row()
+        row=0 if row<0 else row
+
+        ss=template.format(fname=fname, index=row)
+        cb = QtWidgets.QApplication.clipboard()
+        cb.clear(mode=cb.Clipboard)
+        cb.setText(ss, mode=cb.Clipboard)
+        msg = QtWidgets.QMessageBox()
+        msg.setIcon(QtWidgets.QMessageBox.Information)
+        msg.setText(
+            "A python template has been copied to the clipboard."
+        )
+        msg.setWindowTitle("Information")
+        msg.setStandardButtons(QtWidgets.QMessageBox.Ok)
```

Comment:
Just FYI, you can use the static function 'QMessageBox.information()' to do this in one line. Your solution is fine as well though.